js_modules/dagster-ui/packages/ui-core/src/app/analytics.tsx

```diff
@@ -52,28 +55,35 @@ export const dummyAnalytics = () => ({
 
 export const useTrackPageView = () => {
   const analytics = useAnalytics();
-  const {path, specificPath} = usePageContext();
+  const match = useRouteMatch();
+  const {pathname: specificPath} = useLocation();
+  const {path} = match;
+
+  const setCurrentPage = useSetRecoilState(currentPageAtom);
 
-  useEffect(() => {
+  useLayoutEffect(() => {
     // Wait briefly to allow redirects.
     const timer = setTimeout(() => {
       analytics.page(path, specificPath);
     }, PAGEVIEW_DELAY);
+    setCurrentPage({path, specificPath});
```

Comment:
This all looks good and I think I'm in favor of the formal React solution. That said, an alternative two line version of this PR would be to set a hidden `window.__routerPath = path` here, and then have `usePageContext` return `{path: window.__routerPath, specificPath: window.location.path}`.